File: solutions/typescript/roman-numerals/1/roman-numerals.ts
export const toRoman = (integer: number): string => {
  // convert the integer to a roman numeral

  let roman = '';
  let num = integer;

  const romanNumList = {
    M: 1000,
    CM: 900,
    D: 500,
    CD: 400,
    C: 100,
    XC: 90,
    L: 50,
    XL: 40,
    X: 10,
    IX: 9,
    V: 5,
    IV: 4,
    I: 1
  };

  for (let key in romanNumList) {
    let value = romanNumList[key as keyof typeof romanNumList];
    roman += key.repeat(Math.floor(num / value));
    num = num % value;
  }

  return roman;
}

File: solutions/go/roman-numerals/1/roman_numerals.go
package romannumerals

import "errors"

func ToRomanNumeral(input int) (string, error) {
	if input <= 0 || input > 3999 {
		return "", errors.New("invalid input")
	}

	// Define Roman Numeral symbols
	romanSymbols := []string{"M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"}

	// Define Roman Numeral values
	romanValues := []int{1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1}

	// Convert integer to Roman Numeral
	var romanNumeral string
	for i := 0; i < len(romanValues); i++ {
		for input >= romanValues[i] {
			romanNumeral += romanSymbols[i]
			input -= romanValues[i]
		}
	}

	return romanNumeral, nil

}
